sql/core/src/test/scala/org/apache/spark/sql/execution/WholeStageCodegenSuite.scala

```diff
@@ -654,4 +654,30 @@ class WholeStageCodegenSuite extends QueryTest with SharedSparkSession
       }
     }
   }
+
+  test("SPARK-35207: Compute hash consistent between -0.0 and 0.0 doubles with Codegen") {
```

Comment:
I think we don't need to add tests here (It's okay just to add tests in `HashExprSuite`.

Reply:
+1, if you use `checkEvaluation`, both codegen and interpreted are checked.

Reply:
very useful function!